cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/WorkflowHttpHandlerTest.java

```diff
@@ -1325,4 +1327,31 @@ public void testWorkflowTokenPut() throws Exception {
     workflowProgramRuns = getProgramRuns(sparkId, ProgramRunStatus.COMPLETED.name());
     Assert.assertEquals(1, workflowProgramRuns.size());
   }
+
+  @Test
+  public void testWorkflowForkFailure() throws Exception {
```

Comment:
I think you should add a comment, briefly describing the scenario you are testing.
Other than that, LGTM.